pydoctor/templatewriter/writer.py

```diff
@@ -95,3 +116,21 @@ def _writeDocsForOne(self, ob, fobj):
         self.written_pages += 1
         ob.system.progress('html', self.written_pages, self.total_pages, 'pages written')
         flattenToFile(fobj, page)
+
+    def _checkTemplatesV(self) -> None:
+        """
+        Issue warnings when custom templates are outdated. 
+        """
```

Comment:
Yes indeed I did not saw this option at first.
I think we should just politely crash pydoctor if a template that are too new is encountered. 
Because the chances are if the version was bumped it's because a new slot was added, and if so, it will crash at the rendering time.
It should be clear in the docs that the users should not increase the template versions themselves.